moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -149,58 +161,94 @@ const ProjectFiles = props => {
 
   const fileTypes = ["", "Funding", "Plans", "Estimates", "Other"];
 
+  // remove the FilePond and s3 added path for display, ex:
+  // 'private/project/65/80_04072022191747_40d4c982e064d0f9_1800halfscofieldridgepwkydesignprint.pdf'
+  const cleanUpFileKey = (str) => str.replace(/^(?:[^_]*_){3}/g, "");
+
   /**
    * Column configuration for <MaterialTable>
    */
   const columns = [
     {
       title: "Name",
       field: "file_name",
-      validate: rowData => {
+      validate: (rowData) => {
         return rowData.file_name.length > 0 ? true : false;
       },
-      render: record => (
-        <Link
-          className={classes.downloadLink}
-          onClick={() => downloadFileAttachment(record?.file_key, token)}
-        >
-          {record?.file_name}
-        </Link>
-      ),
-      editComponent: props => (
+      editComponent: (props) => (
         <TextField
           variant="standard"
           id="file_name"
           name="file_name"
           value={props.value}
-          onChange={e => props.onChange(e.target.value)}
-          helperText="Required" />
+          onChange={(e) => props.onChange(e.target.value)}
+          helperText="Required"
+        />
       ),
     },
+    {
+      title: "File",
+      field: "file_url",
+      validate: (rowData) => {
+        return rowData.file_name.length > 0 ? true : false;
+      },
+      render: (record) => {
+        if (record.file_key) {
+          return (
+            <Link
+              className={classes.downloadLink}
+              onClick={() => downloadFileAttachment(record?.file_key, token)}
+            >
+              {cleanUpFileKey(record?.file_key)}
+            </Link>
+          );
+        }
+        return isValidUrl(record?.file_url) ? (
+          <ExternalLink
+            className={classes.downloadLink}
+            url={record?.file_url}
+            text={record?.file_url}
+          />
+        ) : (
+          // if the user provided file_url is not a valid url, just render the text
+          <Typography className={classes.codeStyle}>
+            {record?.file_url}
+          </Typography>
+        );
+      },
+      editComponent: (props) =>
+        // users cannot edit the file_key, since its provided by the FilePond upload interface
+        props.rowData.file_key ? (
+          <Typography>{cleanUpFileKey(props.rowData.file_key)}</Typography>
+        ) : (
+          <TextField
+            variant="standard"
+            id="file_path"
+            name="file_path"
+            value={props.value}
+            onChange={(e) => props.onChange(e.target.value)}
```

Comment:
looks like you need to apply the same value handling here as you do in the form: trim empty strings and coerce to `null`.